community-build/test/scala/dotty/communitybuild/CommunityBuildTest.scala

```diff
@@ -0,0 +1,167 @@
+package dotty.communitybuild
+
+import java.nio.file._
+import java.io.{PrintWriter, File}
+import java.nio.charset.StandardCharsets.UTF_8
+import org.junit.{Ignore, Test}
+import org.junit.Assert.{assertEquals, fail}
+
+abstract class BuildTest {
+  lazy val communitybuildDir: Path = Paths.get(sys.props("user.dir") + "/community-build/")
+
+  lazy val compilerVersion: String = {
+    val file = communitybuildDir.resolve("dotty-bootstrapped.version")
+    new String(Files.readAllBytes(file), UTF_8)
+  }
+
+  /** Build the given project with the published local compiler and sbt plugin.
+   *
+   *  This test reads the compiler version from community-build/dotty-bootstrapped.version
+   *  and expects community-build/sbt-dotty.sbt to set the compiler plugin.
+   *
+   *  @param project  The project name, should be a git submodule in community-build/
+   *  @param command  The sbt command used to build the project
+   */
+  def test(project: String, command: String): Unit = {
+    def log(msg: String) = println(Console.GREEN + msg + Console.RESET)
+
+    log(s"Building $project with dotty-bootstrapped $compilerVersion...")
+
+    val projectDir = communitybuildDir.resolve("community-projects").resolve(project)
+
+    if (!Files.exists(projectDir.resolve(".git"))) {
+      fail(s"""
+        |
+        |Missing community-build/community-projects/$project submodule. You can initialize this module using
+        |
+        |    git submodule update --init $projectDir
+        |
+        |""".stripMargin)
+    }
+
+    /** Executes shell command, returns false in case of error. */
+    def exec(binary: String, arguments: String*): Int = {
+      val command = binary +: arguments
+      log(command.mkString(" "))
+      val builder = new ProcessBuilder(command: _*).directory(projectDir.toFile).inheritIO()
+      val process = builder.start()
+      val exitCode = process.waitFor()
+      exitCode
+    }
+
+    // Workarounds for https://github.com/sbt/sbt/issues/4395
+    new File(sys.props("user.home") + "/.sbt/1.0/plugins").mkdirs()
+    val pluginFilePath = communitybuildDir.resolve("sbt-dotty.sbt").toAbsolutePath().toString()
+
+    // Run the sbt command with the compiler version and sbt plugin set in the build
+    val arguments = Seq(
+      "-sbt-version", "1.2.7",
+      s"--addPluginSbtFile=$pluginFilePath",
+      "clean",
+      s"++$compilerVersion!",
```

Comment:
Did you see this suggestion: https://github.com/dotty-staging/pdbp/commit/604c06c50358348d6d2c7daa0aaac0045f43b2f9#r32185635 ?